websites/views.py

```diff
@@ -236,19 +239,53 @@ def publish(self, request, name=None):
         return self.publish_version(name, VERSION_LIVE, request)
 
     @action(
-        detail=True, methods=["post"], permission_classes=[HasWebsitePublishPermission]
+        detail=True,
+        methods=["post", "get"],
+        permission_classes=[HasWebsitePublishPermission],
     )
     def unpublish(self, request, name=None):
         """Unpublish the site and trigger the remove-unpublished-sites pipeline"""
         try:
             website = self.get_object()
-
-            Website.objects.filter(pk=website.pk).update(
-                unpublish_status=PUBLISH_STATUS_NOT_STARTED,
-                last_unpublished_by=request.user,
-            )
-            trigger_unpublished_removal(website)
-            return Response(status=200)
+            if request.method == "GET":
+                ocw_www_dependencies = WebsiteContent.objects.filter(
+                    (
+                        Q(type=CONTENT_TYPE_COURSE_LIST)
+                        | Q(type=CONTENT_TYPE_RESOURCE_COLLECTION)
+                    ),
+                    (
+                        Q(metadata__courses__icontains=website.name)
+                        | Q(metadata__resources__content__icontains=website.name)
+                    ),
+                    website__name=settings.ROOT_WEBSITE_NAME,
+                )
+                course_dependencies = WebsiteContent.objects.filter(
+                    type=CONTENT_TYPE_PAGE,
+                    markdown__contains=website.name,
+                )
```

Comment:
One other place (outside of markdown) where cross-course links show up fairly often is the overall course description. For exmaple, in 8.06
- on ocw https://ocw.mit.edu/courses/8-06-quantum-physics-iii-spring-2005/
- on github: https://github.mit.edu/mitocwcontent/8.06-spring-2018/blob/main/data/course.json#L31

It would be nice if we could do a `metadata__contains=website.name` treating metadata as a string instead of json.

Might not be worth worrying about since @pdpinch indicated that this was the lowest priority of the 3 warnings.

Reply:
yeah, let's do it